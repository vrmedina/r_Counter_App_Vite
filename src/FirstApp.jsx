import PropTypes from 'prop-types';

export const FirstApp = ( { title, subTitle, name } ) => {
    return (
    <>
        <h1>{ title }</h1>
        <p>{ subTitle}</p>
        <p>{ name }</p>
    </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number
}

FirstApp.defaultProps = {
    name: 'No me mandaste nombre',
    subTitle: 1010101010,
    //title: 'No me mandaste titulo'
}