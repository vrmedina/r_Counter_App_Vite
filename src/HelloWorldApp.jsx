export const HelloWorldApp = () => {
  return (
    <div>Hello World App</div>
  )
}

// Minimum react component
//export const A=()=><>a</>
