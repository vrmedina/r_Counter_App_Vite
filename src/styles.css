html, body {
    height: 100%;
    margin: 0;  
    padding: 70px;
    font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
    background-color: #1e1e1e;
    color: #fff;
    font-size: 1.3rem;
}

button {
    background-color: #007bff;
    color: #fff;
    border: none;
    border-radius: 5px;
    margin: 0 5px;
    padding: 10px 20px;
    font-size: 1.3rem;
    cursor: pointer;
}

button:hover {
    background-color: #003b7a;
}