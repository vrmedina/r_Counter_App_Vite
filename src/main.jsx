import React, { StrictMode } from 'react';
import ReactDOM from 'react-dom/client';
import { HelloWorldApp } from './HelloWorldApp';

ReactDOM.createRoot(document.getElementById('root')).render(
    // StrictMode is a tool for highlighting potential problems in an application.
    // Like Fragment, StrictMode does not render any visible UI.
    // It activates additional checks and warnings for its descendants.
    <StrictMode>
        <HelloWorldApp />
    </StrictMode>
);
