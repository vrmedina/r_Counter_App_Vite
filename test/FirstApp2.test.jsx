const { render, screen } = require("@testing-library/react")
import { FirstApp } from '../src/FirstApp'

describe('Pruebas en <FirstApp />', () => {
    const title = 'Hola, soy Goku'
    const subTitle = 123456

    test('Debe coincidir con el snapshot', () => {
        const { container } = render(<FirstApp title={ title }/>)
        expect(container).toMatchSnapshot()
    })

    test('Debe mostrar el mensaje Hola, soy Goku', () => {
        render(<FirstApp title={ title }/>)
        expect(screen.getByText(title)).toBeTruthy()
    })

    test('Debe mostrar el titulo en un h1', () => {
        render(<FirstApp title={ title }/>)
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title)
    })

    test('Debe mostrar el subtitulo enviado por props', () => {
        render(<FirstApp title={ title } subTitle={ subTitle }/>)
        expect(screen.getAllByText(subTitle).length).toBe(2)
    })
})
