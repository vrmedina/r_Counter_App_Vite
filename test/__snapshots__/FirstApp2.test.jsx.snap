// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe coincidir con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <p>
    1010101010
  </p>
  <p>
    1010101010
  </p>
  <p>
    No me mandaste nombre
  </p>
</div>
`;
